skops/hub_utils/_hf_hub.py

```diff
@@ -672,7 +672,7 @@ def download(
         dst.rmdir()
 
     cached_folder = snapshot_download(
-        repo_id=repo_id, revision=revision, use_auth_token=token, **kwargs
+        repo_id=repo_id, revision=revision, token=token, **kwargs
```

Comment:
does this make us have to depend on the latest hf lib?

@Wauplin  how can we make this compatible with multiple hub versions? it's very annoying that we can't easily keep our dependency to an older version.

Reply:
I tested locally (with 0.11.1) and it worked.

Reply:
@adrinjalali You can keep `use_auth_token` for the time being.

By default and for future integrations, it's best to use `token` everywhere. But if you want to ensure backward compatibility with previous versions, you can still use `use_auth_token`. If used it will be treated as `token` thanks to the `@validate_hf_hub_args` validator. The idea is to slowly get rid of every `use_auth_token` mention over time. It will not throw a warning message (yet) as it is still too widely used.

Reply:
And yes, it's not specific to `hfh` 0.12 as the switch has been introduced in `hfh 0.11`.

Reply:
If that's the case we should be able to keep it as is, but our CI fails I guess?

> I tested locally (with 0.11.1) and it worked.

But it doesn't work with `0.10`, does it?

Reply:
The problem is that mypy doesn't understand that `use_auth_token` works because it's not in the signature (I assume the `validate_hf_hub_args` decorator does the actual handling). I'll test with 0.10.

Reply:
Doesn't work with 0.10.1. So we could revert to using `use_auth_token` and add a `# type: ignore` I guess.

I wonder why hf_hub uses this unconventional way of dealing with deprecated parameters, using a traditional approach would not confuse mypy.

Reply:
I'll create a new PR with the changes.

Reply:
Using `token` with `v0.10` will unfortunately not work no.
About the mypy check, that's true that `@validate_hf_hub_args` does not change the signature. I'm not sure we can easily do that in Python :confused: So the workarounds I see are either:
1. put a `# type! ignore` on snapshot_download calls
2. bump `hfh` to have minimal version `0.11`
3. have some logic in `skops` to pass either `token` or `use_auth_token` based on the `hfh` version (:confused:)

I let you see what you prefer for `skops`. What's weird is that mypy should have already complained when migrating to `hfh 0.11`.